packages/@uppy/provider-views/src/SearchFilterInput.jsx

```diff
@@ -0,0 +1,102 @@
+import { h, Fragment } from 'preact'
+import { useEffect, useState, useCallback } from 'preact/hooks'
+import { nanoid } from 'nanoid/non-secure'
+// import debounce from 'lodash.debounce'
+
+export default (props) => {
+  const {
+    search,
+    searchOnInput,
+    searchTerm,
+    showButton,
+    inputLabel,
+    clearSearchLabel,
+    buttonLabel,
+    clearSearch,
+    inputCSSClassName,
+    buttonCSSClassName,
+  } = props
+  const [searchText, setSearchText] = useState(searchTerm ?? '')
+  // const debouncedSearch = debounce((q) => search(q), 1000)
+
+  const validateAndSearch = useCallback((ev) => {
+    ev.preventDefault()
+    search(searchText)
+  }, [search, searchText])
+
+  const handleInput = useCallback((ev) => {
+    const inputValue = ev.target.value
+    setSearchText(inputValue)
+    if (searchOnInput) search(inputValue)
+  }, [setSearchText, searchOnInput, search])
+
+  const handleReset = () => {
+    setSearchText('')
+    if (clearSearch) clearSearch()
+  }
+
+  const [form] = useState(() => {
+    const formEl = document.createElement('form')
+    formEl.setAttribute('tabindex', '-1')
+    formEl.id = nanoid()
+    return formEl
+  })
+
+  useEffect(() => {
+    document.body.appendChild(form)
+    return () => document.body.removeChild(form)
+  }, [form])
+
+  useEffect(() => {
+    form.addEventListener('submit', validateAndSearch)
+    return () => form.removeEventListener('submit', validateAndSearch)
+  }, [form, validateAndSearch])
+
+  return (
+    <Fragment>
+      <input
+        className={`uppy-u-reset ${inputCSSClassName}`}
```

Comment:
Isn't the `CSS` part of this a bit unnecessary? `inputClassName` would be clear enough no?

Reply:
Probably. I guess I tought might be confusing with JS `class` but unlikely in this case.